### PR TITLE
Navigation Bar Component

### LINKED ISSUE #41: Frontend Page Skeleton Component
## Summary
- We need a skeleton for the homepage which will contain all the other homepage components (ie. log records, filter/search options, various buttons)
- This skeleton page should contain the header that allows you to switch pages
- Should be easy to add components to the page

image

## Goals
- [ ] Easily modifiable homepage
- [ ] Contains a header component that matches design specs on Figma
- [ ] Clicking buttons on header switches to new pages
## Implementation
- Should be a pretty straightforward react component

### PR DESCRIPTION
**Pull request by safchow**
## Notion ticket link

[Frontend Page Skeleton Component](https://github.com/uwblueprint/supportive-housing/issues/41)

## Implementation description
* Created a Navigation Bar component
* Created new pages (that are empty), "Resident Directory" and "Employee Directory"
* Updated Homepage to include Log Records table
* Removed auth requirement and email verification from logout so that users do not need to be verified to log out

## Steps to test
1. Verify the Navigation Bar renders, and is styled correctly according to the Figma
2. Verify all the nav bar buttons work:
    a. Resident Directory takes you to `/resident-directory` endpoint
    b. Employee Directory takes you to `/employee-directory` endpoint
    c. Home takes you to `/` endpoint
    d. Logout signs you out, and takes you the sign out page 
    
Note: In testing, I would often run into I/O errors which would cause the page to not render. If you run into this issue, the (painful) solution is to rebuild the deployment:
```
docker compose down 
docker rm -f $(docker ps -a -q) # deletes all running/stopped containers
docker compose down --volumes # brings down all volumes (incl database)
docker volume rm $(docker volume ls -q) # deletes all docker volumes
docker-compose up --build
docker exec -it SHOW-backend /bin/bash -c "flask db upgrade"
```

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [x] If I have made API changes, I have updated the [REST API Docs](https://www.notion.so/uwblueprintexecs/REST-Endpoints-05ce60312bb943439dfda42bb1318536)
- [x] IF I have made changes to the db/models, I have updated the [Data Models Page](https://www.notion.so/uwblueprintexecs/Data-Models-760f8aa06b244eb0842c079ad77987b0)
- [x] I have updated other Docs as needed

### CODE REVIEW

File: backend/app/services/implementations/auth_service.py [hunk 1]
```diff
@@ -184,8 +184,7 @@ def is_authorized_by_user_id(self, access_token, requested_user_id):
             token_user_id = self.user_service.get_user_id_by_auth_id(
                 decoded_id_token["uid"]
             )
-            firebase_user = firebase_admin.auth.get_user(decoded_id_token["uid"])
-            return firebase_user.email_verified and token_user_id == requested_user_id
+            return token_user_id == requested_user_id
```
Comment (DinuWije) [inline backend/app/services/implementations/auth_service.py]: revert this change

**Review by connor-bechthold — APPROVED** [review]
Looks good and works! Only thing I would look into doing is seeing if you can instantiate the nav component somewhere globally so we don't need to add it to every page...that way if someone new comes along and wants to add a new page they don't need to remember to do that. More a nice-to-have though, so I wouldn't say it's necessary in order to merge.

**Review by DinuWije — APPROVED** [review]
LGTM (Let's Go To Mars 🚀)
